### PR TITLE
Allow to specify bind ip address of puppet master

### PR DESCRIPTION
**Pull request by roidelapluie**

### CODE REVIEW

**Comment by mmoll** [comment]
please rebase

**Comment by roidelapluie** [comment]
Done

File: manifests/params.pp [hunk 1]
```diff
@@ -7,6 +7,7 @@
   $version             = 'present'
   $user                = 'puppet'
   $group               = 'puppet'
+  $ip                  = undef
```
Comment (domcleal) [inline manifests/params.pp]: Should this be 'server_ip' for consistency with the class parameter name?

**Comment by domcleal** [comment]
One question, otherwise :+1: 

We should probably check about setting the bind address on Puppet Server (and possibly the master webrick service) some time.

**Comment by ekohl** [comment]
So I recall puppetlabs-apache does not like mixing IPs and names, but since it's undef it's fine. :+1: once the consistency issue is solved.

File: manifests/params.pp (hunk 1 above)
Comment (roidelapluie) [inline manifests/params.pp]: No, @domcleal 
```

$server_dir = $puppet::params::dir,
+ $server_ip = $puppet::params::ip,
$server_port = $puppet::params::port,
```

Comment (roidelapluie) [inline manifests/params.pp]: or you have to modify the whole params.pp?

Comment (domcleal) [inline manifests/params.pp]: Fair point!

**Comment by domcleal** [comment]
I guess we'll overlook the inconsistency for now?

**Comment by mmoll** [comment]
it's consistently "wrong", so :+1:

**Comment by domcleal** [comment]
@ekohl does that mean this isn't going to work as-is?  IIRC this is what I hit when I tried the puppet-foreman patch: https://github.com/theforeman/puppet-foreman/pull/255#issuecomment-70073734

**Comment by ekohl** [comment]
@domcleal as-is it shouldn't be any different from the current code, so I think this works. It may error out in unexpected ways for users.

**Comment by domcleal** [comment]
I think we should just give it a test before merging, to check it doesn't fail like puppet-foreman..

**Comment by ekohl** [comment]
I don't see why puppet-foreman would fail on that PR, so more testing would be very good.

**Comment by roidelapluie** [comment]
Closing after inactivity

**Pull request closed without merging**